nutridine/src/screens/macro/MacroPage.js

```diff
@@ -0,0 +1,225 @@
+import React, { useState, useEffect } from "react";
+import {
+  Button,
+  RangeSlider,
+  RangeSliderTrack,
+  RangeSliderFilledTrack,
+  RangeSliderThumb,
+  VStack,
+  Text,
+  HStack,
+  Input,
+  useColorModeValue,
+  Card,
+  CardBody,
+  Box,
+} from "@chakra-ui/react";
+import useMealsByMacros from "../../hooks/useMealsByMacros";
+import { useMeals } from "../../contexts/MealsContext";
+
+function AttributeSliders() {
+  const { setMeals } = useMeals();
+
+  const [calories, setCalories] = useState([200, 500]);
+  const [protein, setProtein] = useState([10, 50]);
+  const [carbs, setCarbs] = useState([20, 100]);
+  const [fat, setFat] = useState([5, 35]);
+
+  const [showSliders, setShowSliders] = useState(true);
+  const [shouldFetch, setShouldFetch] = useState(false);
```

Comment:
If you need to manually trigger a refetch, a better approach is to export the `fetchData` function from the hook. This way, any component consuming the hook doesn't have to provide the additional parameter, and you don't need to include an extra parameter to negate inside the hooks logic

Reply:
I'm getting merge conflicts on my branch after merging this. I can take care of the refactor and update both of our components

Reply:
Looked into this a little closer. When the macro page calls handle submit, it calls _setDesiredMacros_, which are included in the useEffect callback array, therefore already triggering a refetch. This setup does make an initial fetch before the submit button is called, but i think that is okay since no results will ever be returned, and that endpoint has a very large request limit

Reply:
Thanks @johnmck95  Looking into the new comments now

Reply:
For visibility, this has been addressed in #7